src/pages/en/guides/content-collections.md

```diff
@@ -0,0 +1,367 @@
+---
+import { getCollection, renderEntry } from 'astro:content';
+const blogPosts = await getCollection('blog');
+---
+
+{blogPosts.map(async (post) => {
+  const { headings } = await renderEntry(post);
+  const h1 = headings.find(h => h.depth === 1);
+  return <p>{h1}</p>
+})}
+```
+
+### Access injected frontmatter
+
+Astro allows you to [inject frontmatter using remark or rehype plugins.](/en/guides/markdown-content/#example-injecting-frontmatter) You can access these values using the `injectedFrontmatter` property from `renderEntry`:
+
+```astro "{ injectedFrontmatter }"
+---
+import { getCollection, renderEntry } from 'astro:content';
+const blogPosts = await getCollection('blog');
+---
+
+{blogPosts.map(async (post) => {
+  const { injectedFrontmatter } = await renderEntry(post);
+  return <p>{post.data.title} — {injectedFrontmatter.readingTime}</p>
+})}
+```
+
+Assuming `readingTime` was injected ([see our reading time example](/en/guides/markdown-content/#example-calculate-reading-time)), it will be available on the `injectedFrontmatter` object.
+
+<details>
+<summary>**🙋 Why don't `getCollection` and `getEntry` contain these values?**</summary>
+
+The remark and rehype pipelines are only run when your content is **rendered.** This lets `renderEntry` access anything generated by these plugins like injected frontmatter. To stay performant, `getCollection` and `getEntry` do not have this capability.
+
+</details>
+
+## Generating pages from content collections
+
+You can create pages from your content collections using [dynamic routes](/en/core-concepts/routing/#dynamic-routes).
+
+Say you have a `src/content/blog/` collection, and you want to map these entries to `/posts/*` URLs. You can create a `posts/[...slug].astro` route using `getStaticPaths` like so:
+
+```astro "{ slug: entry.slug }"
+---
+// src/pages/posts/[...slug].astro
+import { getCollection } from 'astro:content';
+export async function getStaticPaths() {
+  const blog = await getCollection('blog');
+  // Map collection entries to pages
+  return blog.map(entry => ({
+    // `entry.slug` is the filename of each blog post with `.md` removed
+    params: { slug: entry.slug },
+  }));
+}
+---
+```
+
+This will generate routes for every entry in our collection, mapping each entry's slug to a URL. For example, an entry at `src/content/blog/hello-world.md` will have a slug of `hello-world`. Because this dynamic route is in `src/pages/posts/`, the final URL will be `/posts/hello-world`.
+
+### Generating pages with nested directories
+
+If you have a collection with [nested directories](#organizing-with-nested-directories) (e.g. when organizing your content by locale) this will be reflected in each entry's `slug`. For example, the collection entry `blog/en/intro.md` will have a slug of `en/intro`.
+
+Using [rest parameters in your dynamic routes](/en/core-concepts/routing/#rest-parameters) like in the example above supports mapping nested slugs out-of-the-box.
```

Comment:
``````suggestion
You can create pages based on your content collections using [dynamic routes](/en/core-concepts/routing/#dynamic-routes). 

Use `getCollection()` inside a [`getStaticPaths()`](/en/reference/api-reference/#getstaticpaths) function to query your content entries and provide the `slug` parameter for each page. 

For example, you can dynamically create a page for each entry in a `blog` collection, including nested directories, by creating a `.astro` page with a [rest parameter in its filename](/en/core-concepts/routing/#rest-parameters) to match file paths of any depth.

	```astro "{ slug: entry.slug }"
	---
	// src/pages/posts/[...slug].astro
	import { getCollection } from 'astro:content';
	
	export async function getStaticPaths() {
	  const blog = await getCollection('blog');
	  return blog.map(entry => ({
	    params: { slug: entry.slug },
	  }));
	}
	---
	```

This will generate page routes for every entry in the `blog` collection, mapping each entry’s slug to a URL. For example, an entry at `src/content/blog/hello-world.md` will have a slug of `hello-world` and the collection entry `src/content/blog/en/intro.md` will have a slug of `en/intro`.

Because this dynamic route is in `src/pages/posts/`, the final URLs will be `/posts/hello-world/` and `/posts/en/intro/`.
``````
Since the code example included the rest parameter anyway, I think it's fine to not call out nested entries specifically in another heading. I think this also further establishes that it's *all* the same collection... and you get everything when you query by default. I like discussing the two together, side-by-side, because you don't actually *do* anything different when you have nested directories. So, I don't think we need to call attention to the differenes, as if they were different things to deal with.